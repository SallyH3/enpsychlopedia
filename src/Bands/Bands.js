import React, { Component } from 'react';
import Band from '../Band/Band.js';

export default class Bands extends Component {
  constructor(props) {
    super(props);
    this.state = {
      active: [],
      inactive: []
    }
  }

  assignActiveBands() {
    let activeBands = this.props.genreBands.filter(band => {
        return band.active === true;
      })
    this.setState({active: activeBands})
  }

  assignInactiveBands() {
    let inactiveBands = this.props.genreBands.filter(band => {
        return band.active === false;
      })
    this.setState({inactive: inactiveBands})
  }

  componentWillReceiveProps() {
    this.assignActiveBands();
    this.assignInactiveBands();
  }

  filterAllBands(props) {
    console.log(props)
    return (
      {this.props.genreBands.map((band, index) => (
        <Band key={index}
               bandName={band.name}
               bandImg={band.image}
        />
      ))}
    )
  }

  filterActiveBands() {
    return(
      {this.state.active.map((band, index) => (
        <Band key={index}
                bandName={band.name}
                bandImg={band.image}
            />
          ))}
    )
  }

  filterInactiveBands() {
    return(
      {this.state.inactive.map((band, index) => (
        <Band key={index}
               bandName={band.name}
               bandImg={band.image}
        />
      ))}
    )
  }

  render() { 
  let fallback = this.props.currentSelection;
  if (!fallback) { 
    return (
      <h1>Hi Sally!</h1>
    )
  }
    return (
      <section className='bands-wrapper'>
        if(this.props.currentSelection === 'all') {
          {this.filterAllBands(this.props)}
        } else if (this.props.currentSelection === 'active') {
          {this.filterActiveBands()}
        } else {
          {this.filterInactiveBands()}
        }
      </section>
    )
  }
}