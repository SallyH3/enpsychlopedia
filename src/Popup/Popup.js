import React from 'react';

const Popup = (props) => {
  console.log(props.currentBandInfo)
  return (
    <div className='popup-container'>
      <img src={props.currentBandInfo.image} />
      <h4>{props.currentBandInfo.name}</h4>
      <h4>{props.currentBandInfo.origin}</h4>
      <h4>{props.currentBandInfo.members.map(member => {
        return <h2>{member}</h2>
      })}</h4>
      <h4>{props.currentBandInfo.label}</h4>
      <h4>{props.currentBandInfo.sideProjects.map(project => {
        return <h2>{project}</h2>
      })}</h4>
    </div>
  )
}

export default Popup;