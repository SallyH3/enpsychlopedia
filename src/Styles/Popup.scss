.popup-container {
  z-index: 3;
  background-color: pink;
  position: absolute;
}