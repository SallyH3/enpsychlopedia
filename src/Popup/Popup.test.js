import React from 'react';
import Popup from './Popup';
import { shallow } from 'enzyme';
import Bands from '../bands/bands';

let mockBand = {name: "Electric Wizard", origin: "Dorset England", image: "https://i.imgur.com/oZCj9bF.jpg", members: ["Jus Oborn","Liz Buckingham","Clayton Burgess","Simon Poole"], genreId: 2, active: true, label: "Rise Above", sideProjects: ["Ramesses","Sourvein","Iron Monkey","Teeth of Lions","Rule the Divine","Crippled Black Phoenix","With the Dead"]};

describe('Popup', () => {
  let wrapper;
  let wrapper2;
  beforeEach(() => {
    wrapper2 = shallow(
      <Bands/>
      );
    wrapper2.setState({'showPopup': true});
    wrapper = shallow(
      <Popup
        currentBandInfo={mockBand}
      />
    );
  });
  
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});