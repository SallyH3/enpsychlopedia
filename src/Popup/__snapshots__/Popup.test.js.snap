// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup should match snapshot 1`] = `
<div
  className="popup-container"
>
  <button
    className="close-popup"
  >
    X
  </button>
  <div
    className="popup-head"
  >
    <img
      src="https://i.imgur.com/oZCj9bF.jpg"
    />
    <div
      className="popup-bandname"
    >
      <h1>
        Electric Wizard
      </h1>
      <h4>
        Dorset England
      </h4>
      <h4>
        Label:   
        Rise Above
      </h4>
    </div>
  </div>
  <div
    className="popup-info"
  >
    <h4>
      Members:   
      <p
        key="0"
      >
        Jus Oborn
      </p>
      <p
        key="1"
      >
        Liz Buckingham
      </p>
      <p
        key="2"
      >
        Clayton Burgess
      </p>
      <p
        key="3"
      >
        Simon Poole
      </p>
    </h4>
    <h4>
      Side Projects:   
      <p
        key="0"
      >
        Ramesses
      </p>
      <p
        key="1"
      >
        Sourvein
      </p>
      <p
        key="2"
      >
        Iron Monkey
      </p>
      <p
        key="3"
      >
        Teeth of Lions
      </p>
      <p
        key="4"
      >
        Rule the Divine
      </p>
      <p
        key="5"
      >
        Crippled Black Phoenix
      </p>
      <p
        key="6"
      >
        With the Dead
      </p>
    </h4>
  </div>
</div>
`;
